### PR TITLE
Raven db 7873

### PR DESCRIPTION
**Pull request by talweiss1982**

### CODE REVIEW

File: Rachis/Rachis/Transport/HttpRaftRequest.cs [hunk 1]
```diff
@@ -54,8 +57,15 @@ private Task SendRequestInternal(Func<HttpRequestMessage> getRequestMessage)
...
                 var requestMessage = getRequestMessage();
-                Response = await HttpClient.SendAsync(requestMessage, _cancellationToken).ConfigureAwait(false);
-
+                try
+                {
+                    Response = await HttpClient.SendAsync(requestMessage, _cancellationToken).ConfigureAwait(false);
+                }
+                catch (Exception e)
+                {
+                    _log.ErrorException($"Error while sending a request {requestMessage.Method} {requestMessage.RequestUri}\r\n content:{requestMessage.Content}\r\n{requestMessage.Headers}",e);
```
Comment (ayende) [inline Rachis/Rachis/Transport/HttpRaftRequest.cs]: Will this actually give you the actual content?

File: Rachis/Rachis/Transport/HttpTransportSender.cs [hunk 2]
```diff
@@ -256,15 +263,20 @@ public void Send(NodeConnectionInfo dest, CanInstallSnapshotRequest req)
...
                     var reply = await httpResponseMessage.Content.ReadAsStringAsync().ConfigureAwait(false);
                     if (httpResponseMessage.IsSuccessStatusCode == false && httpResponseMessage.StatusCode != HttpStatusCode.NotAcceptable)
                     {
-                        _log.Warn("Error checking if can install snapshot to {0}. Status: {1}\r\n{2}", dest.Name, httpResponseMessage.StatusCode, reply);
+                        _log.Warn("Error checking if can install snapshot to {0}. Status: {1}\r\n{2}\r\nreason:{3}", dest.Name, httpResponseMessage.StatusCode, reply,httpResponseMessage.ReasonPhrase);
+                        return;
+                    }
+                    if (httpResponseMessage.StatusCode == HttpStatusCode.NotAcceptable)
+                    {
+                        _log.Warn("Error checking if can install snapshot to {0}. Status: NotAcceptable\r\nreason:{1}", dest.Name, httpResponseMessage.ReasonPhrase);
```
Comment (ayende) [inline Rachis/Rachis/Transport/HttpTransportSender.cs]: Add the reply content here as well

File: Rachis/Rachis/Transport/HttpTransportSender.cs [hunk 3]
```diff
@@ -275,22 +287,26 @@ public void Send(NodeConnectionInfo dest, CanInstallSnapshotRequest req)
...
 
                     var reply = await httpResponseMessage.Content.ReadAsStringAsync().ConfigureAwait(false);
                     if (httpResponseMessage.IsSuccessStatusCode == false && httpResponseMessage.StatusCode != HttpStatusCode.NotAcceptable)
                     {
-                        _log.Warn("Error requesting vote from {0}. Status: {1}\r\n{2}", dest.Name, httpResponseMessage.StatusCode, reply);
+                        _log.Warn("Error requesting vote from {0}. Status: {1}\r\n{2}\r\nreason:{3}", dest.Name, httpResponseMessage.StatusCode, reply, httpResponseMessage.ReasonPhrase);
+                        return;
+                    } else if (httpResponseMessage.StatusCode == HttpStatusCode.NotAcceptable)
+                    {
+                        _log.Warn("Error requesting vote from {0}. Status: NotAcceptable\r\nreason:{1}", dest.Name, httpResponseMessage.ReasonPhrase);
```
Comment (ayende) [inline Rachis/Rachis/Transport/HttpTransportSender.cs]: Add reply here too

File: Rachis/Rachis/Transport/HttpTransportSender.cs [hunk 4]
```diff
@@ -309,15 +325,20 @@ public void Send(NodeConnectionInfo dest, TimeoutNowRequest req)
...
                     var reply = await httpResponseMessage.Content.ReadAsStringAsync().ConfigureAwait(false);
                     if (httpResponseMessage.IsSuccessStatusCode == false)
                     {
-                        _log.Warn("Error appending entries to {0}. Status: {1}\r\n{2}", dest.Name, httpResponseMessage.StatusCode, httpResponseMessage, reply);
+                        _log.Warn("Error sending timeout now to {0}. Status: {1}\r\n{2}", dest.Name, httpResponseMessage.StatusCode, httpResponseMessage, reply);
+                        return;
+                    }
+                    if (httpResponseMessage.StatusCode == HttpStatusCode.NotAcceptable)
+                    {
+                        _log.Warn("Error sending timeout now to {0}. Status: NotAcceptable\r\nreason:{1}", dest.Name, httpResponseMessage.ReasonPhrase);
```
Comment (ayende) [inline Rachis/Rachis/Transport/HttpTransportSender.cs]: Add reply here too

File: Rachis/Rachis/Transport/HttpTransportSender.cs [hunk 5]
```diff
@@ -330,17 +351,23 @@ public void Send(NodeConnectionInfo dest, DisconnectedFromCluster req)
...
                     var reply = await httpResponseMessage.Content.ReadAsStringAsync().ConfigureAwait(false);
                     if (httpResponseMessage.IsSuccessStatusCode == false)
                     {
-                        _log.Warn("Error sending disconnecton notification to {0}. Status: {1}\r\n{2}", dest.Name, httpResponseMessage.StatusCode, httpResponseMessage, reply);
+                        _log.Warn("Error sending disconnecton notification to {0}. Status: {1}\r\n{2}\r\nreason:{3}", dest.Name, httpResponseMessage.StatusCode, httpResponseMessage, reply,httpResponseMessage.ReasonPhrase);
                         return;
                     }
+                    if (httpResponseMessage.StatusCode == HttpStatusCode.NotAcceptable)
+                    {
+                        _log.Warn("Error sending disconnecton notification to {0}. Status: NotAcceptable\r\nreason:{1}", dest.Name, httpResponseMessage.ReasonPhrase);
```
Comment (ayende) [inline Rachis/Rachis/Transport/HttpTransportSender.cs]: Add reply here too

**Comment by talweiss1982** [comment]
Added catch exception in event loop + logging
Added ToString to custom content

**Pull request merged**